### PR TITLE
drive state enum's, idiomatic swift for charging states.

### PR DESCRIPTION
**Pull request by ericlewis**
I researched this on my own tesla with all available shift types, this is comprehensive now. I also updated `ChargeState` to reflect idiomatic Swift. Both of these changes are considered breaking changes and fairly easy to update if adoption is chosen.

### CODE REVIEW

File: Sources/TeslaSwift/Model/ChargeState.swift [hunk 1]
```diff
@@ -11,11 +11,10 @@ import Foundation
...
 	public enum ChargingState: String, Codable {
-		// Caps on purpose to avoid codingkeys
-		case Complete
-		case Charging
-		case Disconnected
-		case Stopped
+		case complete = "Complete"
+		case charging = "Charging"
+		case disconnected = "Disconnected"
+		case stopped = "Stopped"
```
Comment (ericlewis) [inline Sources/TeslaSwift/Model/ChargeState.swift:17]: there isn't so many state, and idiomatic swift is best

File: Sources/TeslaSwift/Model/DriveState.swift [hunk 2]
```diff
@@ -10,8 +10,14 @@ import Foundation
 import CoreLocation
 
 open class DriveState: Codable {
+    public enum ShiftState: String, Codable {
+        case drive = "D"
+        case parked = "P"
+        case reverse = "R"
+        case neutral = "N"
```
Comment (ericlewis) [inline Sources/TeslaSwift/Model/DriveState.swift:17]: I would be happy to provide a method or var on this type to also output a `title`

File: Sources/TeslaSwift/Model/DriveState.swift [hunk 3]
```diff
@@ -10,8 +10,14 @@ import Foundation
...
+    }
 	
-	open var shiftState: String?
+	open var shiftState: ShiftState?
```
Comment (ericlewis) [inline Sources/TeslaSwift/Model/DriveState.swift:20]: breaking change

Comment (ericlewis) [reply Sources/TeslaSwift/Model/DriveState.swift:20]: alternatively, we could keep `shiftState` and introduce `state` which is just `ShiftState`

File: Sources/TeslaSwift/TeslaSwift.swift [hunk 4]
```diff
@@ -160,7 +160,7 @@ open class TeslaSwift {
 		}
 	}
 	
-	open fileprivate(set) var token: AuthToken?
```
Comment (ericlewis) [inline Sources/TeslaSwift/TeslaSwift.swift:163]: spacing.

**Comment by ericlewis** [comment]
> What about the reuse token method? Isn't it enough?

Sorry, I didn't actually mean to add this to the PR. I have an example of how to persist the token using Keychain, but in order to do so currently, it is a bit of a pain.

File: Sources/TeslaSwift/Model/DriveState.swift [hunk 5]
```diff
@@ -10,8 +10,14 @@ import Foundation
 import CoreLocation
 
 open class DriveState: Codable {
+    public enum ShiftState: String, Codable {
+        case drive = "D"
+        case parked = "P"
```
Comment (ericlewis) [inline Sources/TeslaSwift/Model/DriveState.swift:15]:
```suggestion
        case park = "P"
```

File: Sources/TeslaSwift/Model/DriveState.swift (hunk 3 above)
Comment (jonasman) [reply Sources/TeslaSwift/Model/DriveState.swift:20]: This is fine

**Pull request merged**